crux_core/src/testing.rs

```diff
@@ -115,51 +113,7 @@ impl<Ef, Ev> AppContext<Ef, Ev> {
 #[derive(Debug)]
 pub struct Update<Ef, Ev> {
     /// Effects requested from the update run
-    pub effects: Vec<TestEffect<Ef, Ev>>,
+    pub effects: Vec<Ef>,
     /// Events dispatched from the update run
     pub events: Vec<Ev>,
 }
-
-pub struct TestEffect<Ef, Ev> {
-    request: Request<Ef>,
-    context: Rc<AppContext<Ef, Ev>>,
-}
-
-impl<Ef, Ev> TestEffect<Ef, Ev> {
-    pub fn resolve<T>(&self, result: &T) -> Update<Ef, Ev>
-    where
-        T: serde::ser::Serialize,
-    {
-        self.context.steps.resume(
-            self.request.uuid.as_slice(),
-            &bcs::to_bytes(result).unwrap(),
-        );
-        self.context.updates()
-    }
-}
```

Comment:
`TestEffect` is not really needed any more. And with that change the AppTester is starting to look _a lot_ like the Core, to the point that I wonder whether its still useful 🤔